pkg/proxy/provider/http/http.go

```diff
@@ -0,0 +1,164 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/containous/traefik/v2/pkg/config/dynamic"
+	"github.com/containous/traefik/v2/pkg/job"
+	"github.com/containous/traefik/v2/pkg/log"
+	"github.com/containous/traefik/v2/pkg/provider"
+	"github.com/containous/traefik/v2/pkg/safe"
+	"github.com/containous/traefik/v2/pkg/types"
+)
+
+var _ provider.Provider = (*Provider)(nil)
+
+// Provider is a provider.Provider implementation that queries an endpoint for a configuration.
+type Provider struct {
+	Endpoint     string         `description:"Load configuration from this endpoint." json:"endpoint" toml:"endpoint" yaml:"endpoint" export:"true"`
+	PollInterval types.Duration `description:"Polling interval for endpoint." json:"pollInterval,omitempty" toml:"pollInterval,omitempty" yaml:"pollInterval,omitempty"`
+	PollTimeout  types.Duration `description:"Polling timeout for endpoint." json:"pollTimeout,omitempty" toml:"pollTimeout,omitempty" yaml:"pollTimeout,omitempty"`
+}
+
+// New creates a new instance of the HTTP provider.
+func New(endpoint string, pollInterval types.Duration, pollTimeout types.Duration) *Provider {
+	return &Provider{
+		Endpoint:     endpoint,
+		PollInterval: pollInterval,
+		PollTimeout:  pollTimeout,
+	}
+}
+
+// Init the provider.
+func (p *Provider) Init() error {
+	if len(p.Endpoint) == 0 {
+		return fmt.Errorf("a non-empty endpoint is required")
+	}
+
+	if p.PollInterval == 0 {
+		p.PollInterval = types.Duration(15 * time.Second)
+	}
+
+	if p.PollTimeout == 0 {
+		p.PollTimeout = types.Duration(15 * time.Second)
+	}
+
+	return nil
+}
+
+// Provide allows the provider to provide configurations to traefik
+// using the given configuration channel.
+func (p *Provider) Provide(configurationChan chan<- dynamic.Message, pool *safe.Pool) error {
+	pool.GoCtx(func(routineCtx context.Context) {
+		ctxLog := log.With(routineCtx, log.Str(log.ProviderName, "http"))
```

Comment:
WDYT of having a const called `ProviderName` that contains http? We're passing it in a few places here, it could be cleaner.